### PR TITLE
addpkg(x11/alacritty): 0.13.2

### LINKED ISSUE #10148: [Package]: alacritty
### Package description

Alacritty Terminal

### Home page URL

https://github.com/alacritty/alacritty

### Source code URL

https://github.com/alacritty/alacritty

### Packaging policy acknowledgement

- [X] I have read and understand the [Packaging Policy](https://github.com/termux/x11-packages/blob/master/CONTRIBUTING.md#packaging-policy).

### Additional information

I used the Rust compiler(rustc) from the Termux repositories to try to compile alacritty. But without success :(

I followed the steps of the official alacritty documentation

### PR DESCRIPTION
**Pull request by TomJo2000**
closes #10148

Doesn't build yet, but I'm done working on it for now.
If anyone else wants to take a look,
the `android-activity` crate is causing the build failure.
And I'm not sure how to resolve it.

### CODE REVIEW

**Comment by licy183** [comment]
I must go to sleep now. It's about 4:00 am UTC+8. 😭

File: x11-packages/alacritty/build.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
+TERMUX_PKG_HOMEPAGE=https://alacritty.org/
+TERMUX_PKG_DESCRIPTION="A fast, cross-platform, OpenGL terminal emulator"
+TERMUX_PKG_LICENSE="Apache-2.0, MIT"
+TERMUX_PKG_MAINTAINER="Joshua Kahn @TomJo2000"
+# Keep in sync with packages/ncurses/build.sh
+TERMUX_PKG_VERSION=0.13.2
+TERMUX_PKG_SRCURL=https://github.com/alacritty/alacritty/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=e9a54aabc92bbdc25ab1659c2e5a1e9b76f27d101342c8219cc98a730fd46d90
+TERMUX_PKG_DEPENDS="fontconfig, freetype, libxi, libxcursor, libxrandr"
+TERMUX_PKG_BUILD_DEPENDS="libxcb, libxkbcommon, ncurses"
```
Comment (TomJo2000) [inline x11-packages/alacritty/build.sh:10]: Don't we need `scdoc` for the man pages?
(Which I hadn't worried about yet)
- https://github.com/alacritty/alacritty/blob/master/INSTALL.md#manual-page

**Comment by TomJo2000** [comment]
Alright, got everything where it needs to go now.
image

Quick checklist:
- [x] Builds locally
- [x] Builds on CI
- [x] Package contents complete
- [x] Tested on device (Fairphone 5, Android 13, Aarch64)

File: x11-packages/alacritty/build.sh [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+termux_step_make() {
+	cargo build --jobs "$TERMUX_MAKE_PROCESSES" --target "$CARGO_TARGET_NAME" --release
+}
+
+termux_step_make_install() {
+	install -Dm755 -t "$TERMUX_PREFIX/bin" "target/$CARGO_TARGET_NAME/release/alacritty"
+
+	# man pages
+	scdoc < extra/man/alacritty.1.scd          | gzip -c | tee "$TERMUX_PREFIX/share/man/man1/alacritty.1.gz" > /dev/null
```
Comment (truboxl) [inline x11-packages/alacritty/build.sh:78]:
```suggestion
	scdoc < extra/man/alacritty.1.scd          | gzip -c > "$TERMUX_PREFIX/share/man/man1/alacritty.1.gz"
```

Comment (TomJo2000) [reply x11-packages/alacritty/build.sh:78]: I was following upstreams instructions for installing the man pages.
https://github.com/alacritty/alacritty/blob/master/INSTALL.md#manual-page

But yeah that makes much more sense.

Comment (TomJo2000) [reply x11-packages/alacritty/build.sh]: > missed one

Did that commit on my phone while waiting for the bus, not surprised I missed something.

**Pull request merged**